docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+```
+
+## Preventing Regressions
+
+Preventing regressions is one of the fundamental concepts of our performance culture.
+
+**Before introducing any changes**, you should run the benchmarks that test the performance of the feature that you are going to work on and store the results in a **dedicated** folder.
+
+```cmd
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\before" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+Please try to **avoid running any resource-heavy processes** that could **spoil** the benchmark results while running the benchmarks.
+
+You can also create a **copy** of the folder with CoreRun and all the libraries to be able to run the benchmarks against the **unmodified base** in the future.
+
+After you introduce the changes and rebuild the part of CoreFX that you are working on **in Release** you should re-run the benchmarks. Remember to store the results in a different folder.
+
+```cmd
+C:\Projects\corefx\src\System.IO.Pipes\src> dotnet msbuild /p:ConfigurationGroup=Release
+
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\after" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+When you have the results you should use [ResultsComparer](../src/tools/ResultsComparer/README.md) to find out how your changes have affected the performance:
+
+```cmd
+C:\Projects\performance\src\tools\ResultsComparer> dotnet run --base "C:\results\before" --diff "C:\results\after" --threshold 5%
+```
+
+Sample output:
+
+```log
+No Slower results for the provided threshold = 5% and noise filter = 0.3ns.
+```
+
+| Faster                                                                           | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
+| -------------------------------------------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
+| System.IO.Pipes.Tests.Perf_NamedPipeStream_ServerIn_ClientOut.ReadWrite(size: 10 |      1.16 |        297167.47 |        255575.49 |         |
+
+## Solving Regressions
+
+### Repro Case
+
+Once a regression is spotted, the first thing that you need to do is to create a benchmark that shows the problem. Typically every performance bug report comes with a small repro case. This is a perfect candidate for the benchmark (it might require some cleanup).
+
+The next step is to send a PR to this repository with the aforementioned benchmark. Our automation is going to run this benchmark and export the results to our reporting system. When your fix to CoreFX gets merged, our reports are going to show the difference. It also helps us to keep track of the old performance bugs and make sure that they never come back.
```

Comment:
`our reporting system` is this private or public?Is there an available link?

Reply:
As of today, it's private, but @DrewScoggins is working on a new one which is going to be public.